### PR TITLE
Switch current code to use template + misc. changes

### PR DESCRIPTION
**Pull request by azurerazor**
### Summary
Shortens template and changes existing code to use template.
Also includes a quick change on TwoColor.h to use ! instead of ^1

### Submission Link/Screenshots
image
image

### Checklist
- [x] AC on at least 1 problem
- [x] Using tabs for indentation
- [x] Longest line at most 63 characters (with tab length 2)

### CODE REVIEW

**Comment by azurerazor** [comment]
might close this until it can actually pass checks

File: content/data-structures/PS2D.h [hunk 1]
```diff
@@ -11,19 +11,19 @@
...
-	PS2D(vector<vector<long long>>& a) {
-		n = size(a), m = size(a[0]);
-		ps = vector(n+1, vector<long long>(m+1));
+	PS2D(vector<vector<ll>>& a) {
+		n = sz(a), m = sz(a[0]);
+		ps = vector(n+1, vector<ll>(m+1));
 		for (int i = 0; i < n; i++) {
 			for (int j = 0; j < m; j++) {
 				ps[i+1][j+1] = a[i][j] + ps[i][j+1]
 				 + ps[i+1][j] - ps[i][j];
```
Comment (enwask) [inline content/data-structures/PS2D.h]: is this a space before the plus sign or a tab character? doesnt look like a tab

File: content/graph/TwoColor.h [hunk 2]
```diff
@@ -22,7 +22,7 @@ vector<int> TwoColor(vector<vector<int>>& graph) {
 			for (auto v : graph[u]) {
```
Comment (enwask) [inline content/graph/TwoColor.h]: why auto? `int v` is shorter

Comment (enwask) [inline content/graph/TwoColor.h]: also `while (sz(q))`

File: content/data-structures/PS2D.h (hunk 1 above)
Comment (azurerazor) [reply content/data-structures/PS2D.h]: it is a space but I can remove it if necessary

File: content/graph/TwoColor.h (hunk 2 above)
Comment (azurerazor) [reply content/graph/TwoColor.h]: fixed

Comment (azurerazor) [reply content/graph/TwoColor.h]: fixed

File: content/data-structures/PS2D.h (hunk 1 above)
Comment (azurerazor) [reply content/data-structures/PS2D.h]: removed

File: content/data-structures/PS2D.h [hunk 3]
```diff
@@ -11,19 +11,19 @@
...
-		n = size(a), m = size(a[0]);
-		ps = vector(n+1, vector<long long>(m+1));
+	PS2D(vector<vector<ll>>& a) {
+		n = sz(a), m = sz(a[0]);
+		ps = vector(n+1, vector<ll>(m+1));
 		for (int i = 0; i < n; i++) {
 			for (int j = 0; j < m; j++) {
 				ps[i+1][j+1] = a[i][j] + ps[i][j+1]
-				 + ps[i+1][j] - ps[i][j];
+				+ ps[i+1][j] - ps[i][j];
```
Comment (enwask) [inline content/data-structures/PS2D.h]: There should still be indentation here, just not a single space char

**Review by enwask — APPROVED** [review]

**Pull request merged**